src/components/__tests__/components/ResetPassword.test.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import renderer from 'react-test-renderer';
+import configureMockStore from 'redux-mock-store';
+
+import {
+  ResetPassword,
+  mapStateToProps,
+  mapDispatchToProps,
+} from '../../ForgotPassword/ResetPassword';
+
+let wrapper;
+let store;
+const mockFn = jest.fn();
+
+const props = {
+  onSubmit: jest.fn().mockImplementation(() => Promise.resolve({ status: 200 })),
+  onInputChange: mockFn,
+  history: { push: mockFn },
+  match: {
+    params: {
+      userId: 'b2d3f3d8-5893-47df-b715-6f10f451bf92',
+      resetCode: '0382040a-f609-49b6-a43a-f1878ae1b5fd',
+    },
+  },
+};
+const mockStore = configureMockStore();
+
+const defaultState = {
+  validPassword: true,
+  passwordError: 'Password must have at least 6 characters',
+};
+describe('<ResetPassword />', () => {
+  test('should render the <ResetPassword />', () => {
+    const renderedValue = renderer.create(<ResetPassword {...props} />).toJSON();
+    expect(renderedValue).toMatchSnapshot();
+  });
+  test('should render default state', () => {
+    wrapper = shallow(<ResetPassword {...props} />);
+    expect(wrapper.state()).toEqual(defaultState);
+  });
+
+  describe('when clicking on submit button', () => {
+    beforeEach(() => {
+      wrapper = mount(<ResetPassword {...props} />);
+    });
+    test('should call onSubmitFunction', () => {
+      wrapper.find('Button').simulate('click');
+      expect(wrapper.state()).toEqual({ ...defaultState, validPassword: false });
+    });
+  });
+
+  describe('when typing into the newPassword input', () => {
+    beforeAll(() => {
+      wrapper = mount(<ResetPassword {...props} />);
+      wrapper.find('Input[name="newPassword"]').simulate('change', { target: { value: '12345' } });
+    });
+    test('should update the newPassword prop', () => {
+      expect(wrapper.state()).toEqual(defaultState);
+    });
+    test('should update the newPassword prop', () => {
+      expect(wrapper.state()).toEqual(defaultState);
+    });
+  });
+  describe('when typing into the confirmNewpassword input', () => {
```

Comment:
space this test

Reply:
@kimotho-njoki I have added a line before describe (line 65).